src/main/java/pkmncore/DBManager.java

```diff
@@ -0,0 +1,112 @@
+package pkmncore;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DBManager implements StorageUnit {
+
+    final private String dbURL;
+    final private String username;
+    final private String password;
+    final private String driver;
+
+    public DBManager(String dbURL, String username, String password, String driver) {
+        this.dbURL = dbURL;
+        this.username = username;
+        this.password = password;
+        this.driver = driver;
+    }
+
+    public Connection getConnection() throws PokemonError, SQLException {
+        setDriver();
+        return connectToDB();
+    }
+
+    public void save(String name, String height, String[] abilities) throws PokemonError {
+        savePokemonDetails(name, height, abilities);
+    }
+
+    public List<List<String>> getPokemon() throws PokemonError {
+        return getAllPokemon();
+    }
+
+    private void savePokemonDetails(String name, String height, String[] abilities) throws PokemonError {
+        try {
+            Connection connection = getConnection();
```

Comment:
There is a _very interesting_ repetition here and some methods down the line:
* Create a connection
* Create a statement
* **do something**
* Close the connection
* Catch a SQL exception

The only thing that changes is the **"do something"** and the error in the SQL exception.

Do you think you can come up with a clever abstraction to avoid this duplication? Show it to Rabea and have a chat with her, as I showed her a technique when she was here at Ford.